src/OnChain.ts

```diff
@@ -59,6 +60,111 @@ export const getOnChainTransactions = async ({
   }
 }
 
+///////////
+abstract class PayOnChainClient {
+  client
+
+  static clientPayInstance(): PayOnChainClient {
+    return new LndOnChainClient()
+    // return new BitcoindClient()
+  }
+
+  abstract getBalance(): Promise<number>
+
+  abstract getEstimatedFee(
+    sendTo?: {
+      address: string
+      tokens: number
+    }[],
+  ): Promise<number>
+
+  // return txid
+  abstract sendToAddress(address: string, amount: number): Promise<string>
+
+  abstract getTxnFee(id: string): Promise<number>
+}
+
+class LndOnChainClient extends PayOnChainClient {
```

Comment:
is a highly coupled class with getActiveOnchainLnd. Maybe is better pass the lnd client as parametter (the same for BitcoindClient)

Reply:
The main purpose of the classes at the moment is for ease of development purposes. After the bitcoind client works as a replacement to the lnd client, then we could reconsider the purpose and structure of these classes.

Reply:
but you have only one instance (line 68) and in the same file

Reply:
is this still relevant @dolcalmi ?